File: code/python/bubble_sort.py

def bubble_sort(arr):
    """
    Bubble sort algorithm.
    Args:
        arr: list of integers
    """
    for i in range(len(arr)):
        for j in range(len(arr) - 1 - i):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr


if __name__ == "__main__":
    arr = [5, 2, 4, 6, 1, 3]
    print(bubble_sort(arr))

File: code/go/bubble_sort.go
package main

import "fmt"

func bubbleSort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		for j := 0; j < len(arr) - i - 1; j++ {
			if arr[j] > arr[j + 1] {
				arr[j], arr[j + 1] = arr[j + 1], arr[j]
			}
		}
	}
	return arr
}

func main() {
	arr := []int{5, 2, 4, 6, 1, 3}
	bubbleSort(arr)
	fmt.Println(arr)
}